### PR TITLE
Implement support for reading schemas from Packages

### LINKED ISSUE #18: Support reading schema files from installed packages
Implement fetching schema files from installed packages that have defined their custom schemas for preferences (and possibly for other types of configurations too).

For example, if package `LSP` wants to provide schema for its `LSP.sublime-settings` preferences it can include file `LSP.sublime-settings.schema.json` (or just `LSP.sublime-settings.schema` but then the one developing the schema would have to manually associate that extension with JSON syntax) that this package will find and automatically map to `LSP.sublime-settings` file pattern.

The limitation of that system is that one schema can only be associated with one file pattern.
(There could be a special case for preferences where we would also auto-create patterns with platform suffixes.)

@predragnikolic @rwols Thoughts (especially on naming)?

### PR DESCRIPTION
**Pull request by rchl**
Resolves #18

BTW. Some internal details:
 - The schema ID is generated on reading schemas and has a format like `sublime://settings/[id]` where `id` is just monotonically increasing index which is also accessor to in-memory list. Just thought I'll mention that since we've discussed different approaches but it looks like it can just be as simple as this.
 - ~~I've picked `sublime-package-types.json` name for the schema file but not entirely happy with that. I wanted to use `sublime-package.json` but it appears that then the schema for `package.json` will also apply to it. I've created an issue for that at https://github.com/SchemaStore/schemastore/issues/1206 but not sure if it's the right place. I could update patterns myself but would prefer that to be done at the "source".~~
  Switched to `sublime-package.json` and made a fix to prefix patterns with `/`.

@rwols @predragnikolic @jfcherng @deathaxe

### CODE REVIEW

**Comment by rwols** [comment]
lsp_utils doesn't contain a class called `ApiWrapperInterface` at my local setup?

**Comment by rchl** [comment]
Oh, right. I guess I have to ship that first. :)

**Comment by rchl** [comment]
> lsp_utils doesn't contain a class called `ApiWrapperInterface` at my local setup?

Added in https://github.com/sublimelsp/lsp_utils/pull/21

File: sublime-package-types.json [hunk 1]
```diff
@@ -0,0 +1,111 @@
+{
+  "contributions": {
```
Comment (rchl) [inline sublime-package-types.json:2]: The base schema for LSP packages should ideally be in the LSP package itself but I think there is no way to do that currently. I mean, to contribute a generic, pattern-less schema that can be referenced by other schemas. But should be possible to support fairly easily.

BTW. Creating schemas by hand is a bit of a pain. :)

**Comment by predragnikolic** [comment]
maybe rename `sublime-package-types.json` to 
1. `settings-schema.json` (shorter, but no indication that it has someting to do with LSP) or 
2. `lsp-settings-schema.json`(a bit longer, but it communicates that this file is related to lsp) 
3. `rafalint.json` :)

**Comment by rchl** [comment]
It actually is not specific to LSP. This schema is ST Package specific and any package can provide their own types.

**Comment by rchl** [comment]
To clarify, any package can define `sublime-package-types.json` to contribute a settings schema to its own settings file or any other file. In the future that can be extended to allow contributing schema to other *.sublime-foo files. Since this is a `LSP-json`, its schema is for the `LSP-json.sublime-settings` and thus it's LSP-specific.

The `sublime-package-types.json` doesn't have a nice ring to it but I don't have better suggestions at the moment. I think ideal would be `sublime-package.json` but as I've mentioned before, that is problematic as it picks up the schema for `package.json` (which I'd say is a bug in schema store).

**Comment by predragnikolic** [comment]
Should we just put a `/` in front of `package.json` in `LSP-json/lsp-json-schemas.json`
```diff
  {
    "fileMatch": [
-      "package.json"
+      "/package.json"
    ],
    "uri": "https://json.schemastore.org/package"
  },

```
and go with sublime-package.json?

**Comment by rchl** [comment]
We can change the script to prefix all patterns. I wanted to avoid that but now that I think about it, it's probably fine.

**Comment by predragnikolic** [comment]
Now I stopped getting completions when inside `sublime-package.json` file.
I think that it is because `LSP-json/schemas/sublime-package.json`. That file now should be named differently... `sublime-package-schema.json` and inside that file, the $id should be changed to something else, and the `lsp-json-schemas_extra.json:65` should be updated accordingly. :)

That is something that I did to make it work again, but maybe you have a better solution.

**Pull request merged**